docs/how-tos/hello-world-module.md

```diff
@@ -390,13 +494,197 @@ You don't need to edit any of the validate or configuration methods because you'
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="104" }
+   imgFile, err := os.Open(imagePath)
+   if err != nil {
+     return nil, camera.ImageMetadata{}, errors.New("Error opening image.")
+   }
+   defer imgFile.Close()
+   imgByte, err := ioutil.ReadFile(imagePath)
+   return imgByte, camera.ImageMetadata{}, nil
+   ```
+
+1. Delete the `SubscribeRTP` and `Unsubscribe` methods, since they are not applicable to this camera.
+
+1. You can leave the rest of the functions not implemented, because this module is not meant to return a point cloud (`NextPointCloud`), and does not need to return multiple images simultaneously (`Images`).
+   If this camera returned a camera stream instead of a single static file, we would have implemented `Stream` instead of `Read`.
+
+   However, you do need to edit the return statements to return empty structs that match the API.
+   Edit these methods so they look like this:
+
+   ```go {class="line-numbers linkable-line-numbers" data-start="110" }
+   func (s *helloWorldHelloCamera) NewClientFromConn(ctx context.Context, conn rpc.ClientConn, remoteName string, name resource.Name, logger logging.Logger) (camera.Camera, error) {
+       return nil, errors.New("not implemented")
+   }
+
+   func (s *helloWorldHelloCamera) Stream(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
+       return nil, errors.New("not implemented")
+   }
+
+   func (s *helloWorldHelloCamera) Images(ctx context.Context) ([]camera.NamedImage, resource.ResponseMetadata, error) {
+       return []camera.NamedImage{}, resource.ResponseMetadata{}, errors.New("not implemented")
+   }
+
+   func (s *helloWorldHelloCamera) NextPointCloud(ctx context.Context) (pointcloud.PointCloud, error) {
+       return nil, errors.New("not implemented")
+   }
+
+   func (s *helloWorldHelloCamera) Properties(ctx context.Context) (camera.Properties, error) {
+       return camera.Properties{}, errors.New("not implemented")
+   }
+
+   func (s *helloWorldHelloCamera) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+       return map[string]interface{}{}, errors.New("not implemented")
+   }
+   ```
+
+1. Save the file.
+
+### Implement the sensor API
+
+{{< expand "Click if you are also creating a sensor component" >}}
+
+Now edit the sensor class definition to implement the sensor API.
+You don't need to edit any of the validate or configuration methods because you're not adding any configurable attributes for the sensor model.
+
+1. Add `"math/rand"` to the list of imports in <file>hello-sensor.go</file> for the random number generation.<br><br>
+
+1. Since `errUnimplemented` and `Config` are defined in <file>hello-camera.go</file>, you need to change <file>hello-sensor.go</file> to avoid redeclaring them:<br><br>
+
+   - Delete line 20, `errUnimplemented = errors.New("unimplemented")` from <file>hello-sensor.go</file>.<br><br>
```

Comment:
this is line 16 for me. I think it may be impractical to add the line numbers while things with the generator are still changing a lot

Reply:
Oh you're right it should be 16 or 17 but it depends how many newlines they added to imports. Yes it's hard to maintain but it does make it much  easier to follow if it's right....and if it's off by only a line or two I think it'll be okay. I will change the *Config *sensorConfig to a search-replace but for the others since they're one-offs I don't think it makes sense to?